test/CMakeLists.txt

```diff
@@ -123,6 +122,13 @@ if(NOT (MIOPEN_TEST_GFX900 OR MIOPEN_TEST_GFX906 OR MIOPEN_TEST_GFX908 OR MIOPEN
         set(MIOPEN_TEST_GPU_DETECTION_FAILED TRUE)
     endif()
 endif()
+
+set_var_to_condition(MIOPEN_LIBMLIR_SUPPORTS_GFX103X_DEFAULT ${MIOPEN_USE_MLIR} AND (${rocMLIR_VERSION_FLAT} GREATER 100000000))
+option( MIOPEN_LIBMLIR_SUPPORTS_GFX103X "libMLIR capability to support GFX10. Enables MLIR tests on GFX103X." ${MIOPEN_LIBMLIR_SUPPORTS_GFX103X_DEFAULT})
```

Comment:
@jerryyin Overriding is necessary for WORKAROUND_ISSUE_1913 to work (and may be handy in the future). But please feel free to remove `option()` together with W/A in your PR, if you like.

Reply:
Okay I think I understand why you need the workaround better now. But one thing is still unclear, as it looks like the existing condition will always block any GFX103X stages no matter what. I think `MIOPEN_TEST_MLIR_DEFAULT` is meant to be on when:
 - It isn't a gfx103x testing stage (because they will always be supported), Or
 - It is a gfx103x testing stage AND MLIR supports gfx103x

Translating those to conditionals:
> `set_var_to_condition(MIOPEN_TEST_MLIR_DEFAULT ${MIOPEN_USE_MLIR} AND` 
    `((NOT ${MIOPEN_TEST_GFX103X}) OR (${MIOPEN_TEST_GFX103X} AND ${MIOPEN_LIBMLIR_SUPPORTS_GFX103X})))`

Reply:
`(!a || (a && b))` is equivalent to `(!a || b)` which is equivalent to `!(a && !b)`. The conditional proposed above is equivalent to the conditional at line 129. The construct at line 129:

    `...AND NOT (${MIOPEN_TEST_GFX103X} AND NOT ${MIOPEN_LIBMLIR_SUPPORTS_GFX103X})...`

...should be read as: Let's _enforce_ the result to FALSE if we test on GFX10 _and_ libMLIR does _not_ support GFX10. Otherwise do not change the result.

Reply:
No problem at all ;)